pyerrors/input/openQCD.py

```diff
@@ -343,3 +345,209 @@ def _read_array_openQCD2(fp):
+                    L = Ls[0]
+                    if not (supposed_L == L):
+                        raise Exception("It seems the length given in the header and by you contradict each other")
+                else:
+                    raise Exception("Found more than one spatial length in header!")
+            
+            print('dnms:', dn)
+            print('nn:', nn)
+            print('tmax:', tmax)
+            t = fp.read(8) #header4 wird mit 16 bytes ausgelesen... cmax fehlt also als Angabe
+            eps = struct.unpack('d', t)[0]# das hier ist vllt. tol
+            print('eps:', eps)
+
+            Q = []
+            
+            i = r_meas_start*steps
+            while 0 < 1:
+                t = fp.read(4) #int nt
+                if(len(t) < 4):
+                    break
+                nc = struct.unpack('i',t)[0]
+                #print(nc)
+                if(nc != i):
+                    print(nc)
+                    raise Exception('Config ' + str(i) + ' missing?')
+                else:
+                    t = fp.read(8 * tmax * (nn + 1))#wegwerfen, Wsl
+                    t = fp.read(8 * tmax * (nn + 1))#wegwerfen, Ysl
+                    t = fp.read(8 * tmax * (nn + 1))#NICHT wegwerfen, das ist Qsl, das, was wirr wollen
+                    #unpack the array of Qtops, on each timeslice t=0,...,tmax-1 and the 
+                    #measurement number in = 0...nn (see README.qcd1)
+                    tmpd = struct.unpack('d' * tmax * (nn + 1), t)
+                    Q.append(tmpd)
+                    i += 1*steps
+                #print(tmp)
+                #set step by reading all entries, then set stepsize, then check if everything is there
+                #make a dtr_config param, which is checked against difference...
+                #difference != step
+                #len(set(difference)) == 1
+                #!!!also implement the idl stuff for everything...
+        print(len(Q))
+        print('max_t:', dn * (nn) * eps)
+        
+        t_aim = (c * L) ** 2 / 8
+
+        print('t_aim:', t_aim)
+        index_aim = round(t_aim / eps / dn)
+        print('index_aim:', index_aim)
+
+
+        Q_sum = []
+        for i, item in enumerate(Q):
+            Q_sum.append([sum(item[current:current + tmax]) for current in range(0, len(item), tmax)])
+        print(len(Q_sum))
+        print(len(Q_sum[0]))
+        Q_round = []
+        for i in range(len(Q) // dtr_cnfg):
+            Q_round.append(round(Q_sum[dtr_cnfg * i][index_aim]))
+
+        replica = len(files)
+
+        tmp = []
+        for q in Q_round:
+            if full:
+                tmp.append(q)
+            else:
+                if int(q) == target:
+                    tmp.append(1.0)
+                else:
+                    tmp.append(0.0)
+        
+        truncated_file = file[:-7] #as seen in previous examples, this could lead to some weird behaviour... maybe -7 fixes this.
```

Comment:
Why does `[:-7]` help here?

Reply:
Compared to before (I think it was something like `file.split(".")[0]`) this version is more stable when it comes to filenames containing a ".", e.g. T32L32_b3.80_k0.13... , which appeared in my analyses more than often.

Reply:
On the other hand, of course `[:-7]` is not stable towards changes in the file extension.